src/components/features/forms/MoveGoodsForm.tsx

```diff
@@ -0,0 +1,226 @@
+import { Box } from '@mui/material'
+import { Control, Controller, FieldErrors } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+import TextField from '@mui/material/TextField'
+import Autocomplete from '@mui/material/Autocomplete'
+import { useEffect, useState } from 'react'
+import { NewDeliveryStep4FormData } from '@/schemas/newDeliverySchemas'
+import { useNewDeliveryContext } from '@/hooks/useNewDeliveryContext'
+import DeleteIcon from '@mui/icons-material/Delete'
+
+interface GoodDetailsFormProps {
+  control: Control<NewDeliveryStep4FormData, any>
+  errors: FieldErrors<NewDeliveryStep4FormData>
+  goodType: string[]
+  zones: string[]
+  index: number
+  onDeleteHandler: (goodTypeValue: string | null, quantity: number) => void
+  formsCount: number
+}
+
+export default function MoveGoodsForm({
+  control,
+  errors,
+  goodType,
+  zones,
+  index,
+  onDeleteHandler,
+  formsCount
+}: GoodDetailsFormProps) {
+  const { t: translate } = useTranslation()
+  const { formsData, goodsTypeQuantityStep4, setGoodTypeQuantityStep4, setAlertQuantities } =
+    useNewDeliveryContext()
+
+  const [goodTypeValue, setGoodTypeValue] = useState<string | null>(null)
```

Comment:
we keep the state for saving the current choices (goodType, quantity, zone), but when we do something inside we need to update the `step4` array that we have inside the context. 
So listen for changes and when we have value for something, lets update the state inside step4. You have an index so you can use that, maybe create a function inside the context and export it so that you can use it to update an item based on the index.

Reply:
Fixed